docs/src/developer/spack.md

```diff
@@ -0,0 +1,170 @@
+This is particularly relevant if you're adding new dependencies, changing build
+options, or preparing a new release.
+
+The first step is to add *Palace*'s Spack repository to your Spack
+configuration. Assuming your *Palace* checkout is in `~/repos/palace`, you would
+run:
+
+```sh
+spack repo add ~/repos/palace/spack_repo/local
+```
+
+You should see a confirmation message like `==> Added repo with namespace 'local'.` The namespace `local` is significant because it allows you to
+distinguish between your development version of the package and the official one
+in Spack's built-in repository. When you refer to `local.palace`, you're
+explicitly using the recipe from your personal repository, while
+`builtin.palace` refers to the official recipe that ships with Spack.
+
+To verify that everything is working, try making a small change to the package
+file. For example, you could edit `spack_repo/local/packages/palace/package.py`
+and change the docstring to all capital letters. Then run:
+
+```sh
+spack info local.palace | head -n 4
+```
+
+You should see your modified docstring. Compare this with the output of `spack info builtin.palace | head -n 4` to see the difference between your local
+version and the official one.
+
+One important detail is that the `local` repository takes priority over
+`builtin`. This means that if you just run `spack info palace` without a prefix,
+Spack will use your local version. While this is convenient, we'll continue
+using the explicit `local.` prefix in examples for clarity.
+
+#### Working Around a Spack Bug with Patches
+
+There's currently a known issue in Spack version 1.1.0 that affects packages
+using patches when they come from non-builtin repositories (see the [bug
+report](https://github.com/spack/spack/issues/51505)). Since *Palace* uses
+patches, you'll run into this problem. The workaround is straightforward but a
+bit inelegant: you need to add a copy of the MFEM package to your local
+repository as well.
+
+Rather than maintaining a duplicate copy of the MFEM recipe, you can create a
+symbolic link from Spack's built-in MFEM package to your local repository:
+
+```sh
+ln -s "$(spack location --repo builtin)"/packages/mfem ~/repos/palace/spack_repo/local/packages/
+```
+
+This command uses Spack's `location` subcommand to find where the built-in
+repository lives, then creates a symlink to the MFEM package directory. To
+verify that the symlink was created correctly, run:
+
+```sh
+spack info local.mfem | head -n 4
+```
+
+If you see MFEM's package information, you're all set. This workaround ensures
+that when Spack processes patches for *Palace*, it can find MFEM in the same
+repository namespace and avoid the bug.
+
+## Iterating on Package Changes
+
+With your local repository configured, you can now iterate on *Palace*'s
+`package.py` file. After making changes, you can test them by installing
+*Palace* again. You can also use `spack spec local.palace@develop` to see how
```

Comment:
I think `spack spec` and `spack concretize` would be doing the same thing here, but `spack spec` is nice because you get more / better information IIRC... Not sure if you should change anything here...